framework/src/math_utils.cpp

```diff
@@ -25,6 +25,20 @@ namespace gvk
 		return glm::vec3(m[3][0], m[3][1], m[3][2]);
 	}
 
+	glm::mat4 cancel_translation_from_matrix(const glm::mat4& m)
+	{
+		return glm::mat4(m[0], m[1], m[2], glm::vec4(0.0f, 0.0f, 0.0f, 1.0f));
+	}
+
+	glm::mat4 mirror_matrix(const glm::mat4& m, const glm::length_t axis)
+	{
+		assert(0 <= axis && axis <= 2);
+
+		auto result(m);
+		result[axis] *= -1.0f;
```

Comment:
The value of `enum struct principal_axis` could be directly converted to the index with `static_cast<std::underlying_type<principal_axis>::type>(aAxis)`